### DISCUSSION TITLE
Go language API bindings

### DISCUSSION THREAD

**Discussion by WeiZhang555** [Z3Prover/z3#5025] [Ideas]
Hi, do you have any plans for Golang API bindings? I can find some third part Golang bindings such as https://github.com/mitchellh/go-z3 and github.com/aclements/go-z3/z3 but they are both incomplete and not formal.

Since Golang has been a really popular language, do you plan to add a new Golang binding? It would be quite useful to Golang users(it's a lot as I know.)

Thanks!

**Comment by NikolajBjorner** [comment]
You are welcome to add bindings and a pull request. 
The Travis and Azure DevOps builds should run unit tests similar to other bindings.

**Comment by WeiZhang555** [comment]
@NikolajBjorner OK, I'll have a try, hope I can make some real contributions to Z3 prover 😄

**Comment by NikolajBjorner** [comment]
How did it go?

**Comment by WeiZhang555** [comment]
@NikolajBjorner  Thanks for your interest! :-)

I've made some parts working, and currently I'm working on a new internal project based on the Golang Z3 binding, and keep improving the APIs in this process.

I'll make a pull request once it's completed and get open source permissions from my employer, but I guess it will take some time(maybe several months sadly...)
